### DISCUSSION TITLE
class Meta: incompatibility should be ignored for Django models and similar classes

### DISCUSSION THREAD

**Discussion by Tatsh** [microsoft/pyright#6570] [Ideas]
I do not know the best way to do this but it would be nice to see some way to have this supported. Maybe a setting like `ignoreClassClassMeta`?

```python
class UnmanagedTable(Model):
    class Meta:
        managed = False
```

This currently shows:

```
"Meta" overrides symbol of same name in class "Model"
  "UnmanagedTable.Meta" is incompatible with "django.db.models.base.Model.Meta"
  Type "type[UnmanagedTable.Meta]" cannot be assigned to type "type[django.db.models.base.Model.Meta]"
```

**Comment by erictraut** [comment]
Pyright doesn't have any knowledge of any third-party libraries, and we wouldn't ever add a feature that is specific to django.

From a type checking perspective, this is a legitimate type error. This particular error is reported with the strict-mode `reportIncompatibleVariableOverride` check, which is disabled by default. I presume that you've manually enabled it in your configuration. If you don't want to enforce this type checking rule, you can simply not enable it. Alternatively, you can also use `# type: ignore` if you'd like to suppress the error on a single line.
